src/Cli/Microsoft.TemplateEngine.Cli/CliTemplateParameter.cs

```diff
@@ -150,7 +151,14 @@ internal Option GetOption(IReadOnlySet<string> aliases)
                 if (!string.IsNullOrWhiteSpace(DefaultValue)
                     || (Type == ParameterType.String || Type == ParameterType.Choice) && DefaultValue != null)
                 {
-                    option.SetDefaultValue(DefaultValue);
+                    if (option is Option<string> stringOption)
```

Comment:
`SetDefaultValue` is now avilable only for `Option<T>` (not the non-generic `Option`), to avoid type mismatches (example: `Option<int>` using default value "abc").

I was not 100% sure that it's never something else than `Option<string>` here, so I've added the `Debug.Fail` to throw  during development and not throw in the release builds.